controllers/k8ssandra/stargate.go

```diff
@@ -192,3 +198,104 @@ func (r *K8ssandraClusterReconciler) removeStargateStatus(kc *api.K8ssandraClust
 		}
 	}
 }
+
+func (r *K8ssandraClusterReconciler) reconcileStargateConfigMap(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	desiredConfig map[string]interface{},
+	userConfigMapContent,
+	namespace string,
+	logger logr.Logger,
+) result.ReconcileResult {
+	logger.Info("Reconciling Stargate Cassandra yaml configMap on namespace : " + namespace)
+	filteredCassandraConfig := filterYamlConfig(desiredConfig["cassandra-yaml"].(map[string]interface{}))
+
+	configYamlString := ""
+	if len(filteredCassandraConfig) > 0 {
+		if configYaml, err := yaml.Marshal(filteredCassandraConfig); err != nil {
+			return result.Error(err)
+		} else {
+			configYamlString = string(configYaml)
+		}
+	}
+
+	if userConfigMapContent != "" {
+		separator := "\n"
+		if strings.HasSuffix(userConfigMapContent, "\n") {
+			separator = ""
+		}
+		configYamlString = userConfigMapContent + separator + configYamlString
+	}
+
+	configMapKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      stargate.CassandraConfigMap,
+	}
+
+	logger = logger.WithValues("StargateConfigMap", configMapKey)
+	desiredConfigMap := createStargateConfigMap(namespace, configYamlString, kc)
```

Comment:
We need to add a watch for ConfigMaps. This is missing in main as well. As things are currently in main, I thin we only need to add a watch in the Stargate reconciler. It should probably use a cluster label filter (see [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/k8ssandra/k8ssandracluster_controller.go#L182)).

With these changes, it gets a little confusing. Since the K8ssandraCluster controller is reconciling the generated ConfigMap, I think it also needs a ConfigMap watch. 

This makes me wonder if it would be better for all of this ConfigMap work to be done in the Stargate controller.